TestsCommon/RunSettings.cs

```diff
@@ -103,4 +117,19 @@ private static string InitializeParameter(TestParameters parameters, string para
             return null;
         }
     }
+    protected static bool isPlaceholderValueOrEmpty(string propertyName){
+        return string.IsNullOrEmpty(propertyName) || propertyName.Contains(DashDash);
+    }
+
+    protected static T replaceUnprovidedSettingsValueWithOverride<T>(RunSettings overrideSettings, string propertyName){
```

Comment:
nit: PascalCase and private
```suggestion
    private static T ReplaceUnprovidedSettingsValueWithOverride<T>(RunSettings overrideSettings, string propertyName){
```

Reply:
Wouldn't Protected be better so that the methods can be inherited by child classes?

Reply:
That change can be made when we decide to use it with inheritance, which is not the case at the moment. Keeping it private is self-documenting to understand in which contexts this piece of code is called.